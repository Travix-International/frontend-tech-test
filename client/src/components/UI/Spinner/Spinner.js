import React from 'react'

import styles from './Spinner.scss'

const Spinner = () => (
  <div className={styles.Spinner} data-testid="spinner">
    <div />
    <div />
    <div />
    <div />
  </div>
)

export default Spinner
