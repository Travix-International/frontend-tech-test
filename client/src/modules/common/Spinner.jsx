import React from 'react';

export default class Spinner extends React.Component {
  render() {
    return (
      <i className="fa fa-spinner fa-spin" />
    );
  }
}
