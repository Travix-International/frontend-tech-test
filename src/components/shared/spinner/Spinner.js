import React from 'react'
import PropTypes from 'prop-types'
import { PropagateLoader } from 'react-spinners'
import styles from './spinner.scss'

export default function Spinner({ color, size, loading }) {
  return (
    <div className={styles.spinner}>
      <PropagateLoader
        color={color}
        size={size}
        loading={loading}
      />
    </div>
  )
}

Spinner.propTypes = {
  color: PropTypes.string,
  size: PropTypes.number,
  loading: PropTypes.bool
}

Spinner.defaultProps = {
  color: '#0062cc',
  size: 15,
  loading: true
}
