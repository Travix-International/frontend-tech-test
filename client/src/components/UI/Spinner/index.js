import Spinner from './Spinner.js'

export { Spinner }
