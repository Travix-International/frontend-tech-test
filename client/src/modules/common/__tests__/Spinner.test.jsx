import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Spinner from '../Spinner';


describe('Common.Spinner', () => {
  it('should render with props', () => {
    const spinner = shallow(<Spinner />);

    expect(toJson(spinner)).toMatchSnapshot();
  });
});
