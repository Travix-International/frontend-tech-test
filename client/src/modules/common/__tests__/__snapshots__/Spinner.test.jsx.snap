// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Common.Spinner should render with props 1`] = `
<i
  className="fa fa-spinner fa-spin"
/>
`;
